### PR TITLE
Add Affine Cipher exercise

### PR DESCRIPTION
**Pull request by tofische**
Added a new Haskell practice exercise Affine Cipher based on the available problem specification.

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
@MatthijsBlom Mind taking a look?

File: exercises/practice/affine-cipher/.docs/instructions.append.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Hints
```
Comment (MatthijsBlom) [inline exercises/practice/affine-cipher/.docs/instructions.append.md:1]: Going by [python/.../affine-cipher/.../instructions.append.md](https://github.com/exercism/python/blob/59a459abe8ee5e7f2fe69040d4d020284aa6d1a8/exercises/practice/affine-cipher/.docs/instructions.append.md?plain=1) ([rendered](https://exercism.org/tracks/python/exercises/affine-cipher)), if you wish to show a header then you'll need to add an extra level 2 header.

File: exercises/practice/affine-cipher/.docs/instructions.append.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+You need to implement the `decode` and `encode` functions, which decode and encode a `String` using an Affine cipher.
+You can use the provided signature if you are unsure about the types, but don't let it restrict your creativity.
+
+This exercise works with textual data. For historical reasons, Haskell's
+`String` type is synonymous with `[Char]`, a list of characters. For more
+efficient handling of textual data, the `Text` type can be used.
```
Comment (MatthijsBlom) [inline exercises/practice/affine-cipher/.docs/instructions.append.md:8]:
```suggestion
This exercise works with textual data.
For historical reasons, Haskell's `String` type is synonymous with `[Char]`, a list of characters.
For more efficient handling of textual data, the `Text` type can be used.
```

One sentence per line.

File: exercises/practice/affine-cipher/.docs/instructions.append.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+As an optional extension to this exercise, you can
+
+- read about [string types](https://haskell-lang.org/tutorial/string-types) in
```
Comment (MatthijsBlom) [inline exercises/practice/affine-cipher/.docs/instructions.append.md:12]: haskell-lang.org seems dead. Perhaps the Archive has a copy?

Comment (tofische) [reply exercises/practice/affine-cipher/.docs/instructions.append.md:12]: I think we can remove this link as there is a link to `Data.Text` later on the page, which is IMHO sufficient.

File: exercises/practice/affine-cipher/.docs/instructions.append.md (hunk 2 above)
Comment (tofische) [reply exercises/practice/affine-cipher/.docs/instructions.append.md]: Done.

File: exercises/practice/affine-cipher/.docs/instructions.append.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+This exercise works with textual data.
+For historical reasons, Haskell's `String` type is synonymous with `[Char]`, a list of characters.
+For more efficient handling of textual data, the `Text` type can be used.
+
+As an optional extension to this exercise, you can
+
+- add `- text` to your list of dependencies in package.yaml.
+- import `Data.Text` in [the following
+  way](https://hackernoon.com/4-steps-to-a-better-imports-list-in-haskell-43a3d868273c):
```
Comment (MatthijsBlom) [inline exercises/practice/affine-cipher/.docs/instructions.append.md:14]:
```suggestion
- import `Data.Text` in [the following way](https://hackernoon.com/4-steps-to-a-better-imports-list-in-haskell-43a3d868273c):
```

Sorry, missed some split lines before.

File: exercises/practice/affine-cipher/.docs/instructions.append.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+```haskell
+import qualified Data.Text as T
+import           Data.Text (Text)
+```
+
+- use the `Text` type e.g. `decode :: Text -> Text` and refer to
+  `Data.Text` combinators as e.g. `T.pack`.
+- look up the documentation for
+  [`Data.Text`](https://hackage.haskell.org/package/text/docs/Data-Text.html).
```
Comment (MatthijsBlom) [inline exercises/practice/affine-cipher/.docs/instructions.append.md:24]:
```suggestion
- use the `Text` type e.g. `decode :: Text -> Text` and refer to `Data.Text` combinators as e.g. `T.pack`.
- look up the documentation for [`Data.Text`](https://hackage.haskell.org/package/text/docs/Data-Text.html).
```

File: exercises/practice/affine-cipher/.docs/instructions.append.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+- import `Data.Text` in [the following
+  way](https://hackernoon.com/4-steps-to-a-better-imports-list-in-haskell-43a3d868273c):
+
+```haskell
+import qualified Data.Text as T
+import           Data.Text (Text)
+```
+
+- use the `Text` type e.g. `decode :: Text -> Text` and refer to
+  `Data.Text` combinators as e.g. `T.pack`.
```
Comment (MatthijsBlom) [inline exercises/practice/affine-cipher/.docs/instructions.append.md:22]: `Text.pack` is unnecessary when `OverloadedStrings` is enabled. Do we want to recommend this extension? Though it is idiomatic, it is one more complication as well.

`OverloadedStrings` has been mentioned once before on this track: in [bob/.approaches/text/content.md](https://github.com/exercism/haskell/blob/000e69d1d654691503ebf0e2f1c11664a5d371b6/exercises/practice/bob/.approaches/text/content.md?plain=1#L44-L82).

**Review by MatthijsBlom — COMMENTED** [review]
@ErikSchierboom looks good to me. My one question is not very important.

**Comment by tofische** [comment]
Can we merge this PR or is still something missing?

**Comment by MatthijsBlom** [comment]
@ErikSchierboom is/was stricken by flu.

I'm fine with merging.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**